lib/History.js

```diff
@@ -10,13 +10,25 @@ class History extends Request {
 
   // GET /api/history/period/<timestamp>
   // https://home-assistant.io/developers/rest_api/#get-apihistoryperiodlttimestamp
-  state(timestamp, filter) {
-    if (moment(timestamp, 'YYYY-MM-DDTHH:mm:ssZ', true).isValid() === false) {
+  state(timestamp, filter, endTimestamp) {
+    if (timestamp !== undefined && moment(timestamp, 'YYYY-MM-DDTHH:mm:ssZ', true).isValid() === false) {
       return Promise.reject(new Error('Invalid timestamp format.'));
     }
+    if (endTimestamp !== undefined && moment(endTimestamp, 'YYYY-MM-DDTHH:mm:ssZ', true).isValid() === false) {
+      return Promise.reject(new Error('Invalid timestamp for endTimestamp format.'));
+    }
+
+    const qs = {};
+
+    if (filter !== undefined) {
+      cs.filter_entity_id = filter;
+    }
+
+    if (endTimestamp !== undefined) {
+      cs.end_time = endTimestamp;
+    }
 
-    const qs = filter ? { filter_entity_id: filter } : null;
-    return this._get(`/history/period/${timestamp}`, qs);
+    return this._get(`/history/period/${timestamp || ""}`, qs);
```

Comment:
Lint rules for this project dictate these should be single quotes, not double quotes.

Reply:
Seems like I hadn't any tooling installed back then to show the obvious, thanks for the hint. 👍